lib/backend/etcd/etcd.go

```diff
@@ -46,8 +46,12 @@ type EtcdClient struct {
 	etcdKeysAPI etcd.KeysAPI
 }
 
-func NewEtcdClient(config *capi.EtcdConfig) (*EtcdClient, error) {
-	// Determine the location from the authority or the endpoints.  The endpoints
+func getEtcdEndpoints(config *capi.EtcdConfig) ([]string, error) {
+	if (config.EtcdAuthority != "" || config.EtcdEndpoints != "") && config.EtcdDiscoverySrv != "" {
```

Comment:
I think this will always hit if EtcdDiscoverySrv is set because EtcdAuthority is defaulted.  Right?